src/main/java/com/rocketden/main/service/RoomService.java

```diff
@@ -128,7 +129,30 @@ public RoomDto updateRoomHost(String roomId, UpdateHostRequest request) {
 
         RoomDto roomDto = RoomMapper.toDto(room);
         sendSocketUpdate(roomDto);
-        return RoomMapper.toDto(room);
+        return roomDto;
+    }
+
+    public RoomDto updateRoomSettings(String roomId, UpdateSettingsRequest request) {
+        Room room = repository.findRoomByRoomId(roomId);
+
+        // Return error if room could not be found
+        if (room == null) {
+            throw new ApiException(RoomError.NOT_FOUND);
+        }
+
+        // Return error if the initiator is not the host
+        User initiator = UserMapper.toEntity(request.getInitiator());
+        if (!room.getHost().equals(initiator)) {
+            throw new ApiException(RoomError.INVALID_PERMISSIONS);
+        }
+
+        // Set new difficulty value
+        room.setDifficulty(request.getDifficulty());
+        repository.save(room);
+
+        RoomDto roomDto = RoomMapper.toDto(room);
+        sendSocketUpdate(roomDto);
```

Comment:
As it currently stands, the problem difficulty (if I understand correctly) is not reflected on the frontend, so no update will be triggered from this socket message.

That being said, I'm assuming that will be the next PR step, so I don't really have a problem with this -- just wanted to point it out.